### DISCUSSION TITLE
Issue with Getting Live Gold Prices (XAUUSD)

### DISCUSSION THREAD

**Discussion by brobbins107** [erdewit/ib_insync#506] [General]
Hey,

I am trying to get spot gold prices (XAUUSD) and ib_insync or IB is telling me that the requested market data is not subscribed. I called IB and they said that live data is available without subscription.They suggested it might be an issue with a 3rd party API. I also can see live data in trader workstation. I'm wondering if others can pull live gold?

Requested market data is not subscribed.Delayed market data is available.Error&BEST/CFD/Top&BEST/CFD/Top, contract: CFD(conId=457068913, symbol='XAUUSD', exchange='SMART', currency='USD', localSymbol='XAUUSD', tradingClass='XAUUSD')

**Comment by mattsta** [comment]
Works for me with just `CFD("XAUUSD")` and no other options required. Gives me live results: `XAUUSD   :   1,663.35`

Also for my connection it's using market endpoint `eufarm` for the data:

```haskell
[code 2119]: Market data farm is connecting:eufarm
[code 2104]: Market data farm connection is OK:eufarm
```

**Reply by brobbins107** [reply]
Would you mind sharing your code?  When I enter without an exchange, it asks me for one. 

Here is what I entered:

print('Connecting to IB with Client ID From Config File')
ib = IB()
ib.connect('127.0.0.1', 7497, clientId=18)

contract = CFD(symbol = ticker, exchange = 'SMART')
ib.sleep(2)
print(contract)
m_data = ib.reqMktData(contract) 
ib.sleep(2.0)
print(m_data.last)

The error message I get is Error 354, reqId 3: Requested market data is not subscribed.Delayed market data is available.Error&BEST/CFD/Top&BEST/CFD/Top, contract: CFD(symbol='XAUUSD', exchange='SMART')
nan.

**Reply by mattsta** [reply]
Did you try removing `exchange = 'SMART'` from the contract? It worked for me just using the symbol with no other parameters. I think `SMART` is only for equities and options so using `SMART` with `CFD()` isn't a valid configuration.

IBKR will (usually) fill in good defaults when optional arguments are empty.

**Reply by brobbins107** [reply]
This is what I get without the exchange:

CFD(symbol='XAUUSD')
Error 321, reqId 3: Error validating request.-'bW' : cause - Please enter exchange, contract: CFD(symbol='XAUUSD')
nan

I checked with a few friends who also cannot pull it (all on East coast of US). Despite IB's protestations, I think this is likely an issue on their end, not with ib_insync. I wonder if it has to do with 'IBKR Spot Gold (symbol XAUUSD) and IBKR Spot Silver (symbol XAGUSD) are available to most customers who are not residents of the U.S. or certain other countries.'

**Reply by mattsta** [reply]
oh, interesting. That is odd then. I can't do a fresh test right now because it's the 4 hour Friday IBKR shutdown time, but I'm basically doing this (were you missing the qualification of the contract where IBKR fills in the missing details?):

```python
symbol = "XAUUSD"
contract = CFD(symbol)
self.ib.qualifyContracts(contract)
quote = self.ib.reqMktData(contract)
```

**Reply by brobbins107** [reply]
Yeah it is weird. It auto-populates 'SMART' for me when I qualify the contracts and then tells me I need a subscription. Could I trouble you to specify the exchange and contractID it auto-populates for you? 

Code:

contract = CFD("XAUUSD")
ib.qualifyContracts(contract)
ib.sleep(0.15)
quote= ib.reqMktData(contract)
print(quote)
ib.sleep(1.0)
market_price = quote.marketPrice()
ib.sleep(1.0)

Output:
Connecting to IB with Client ID From Config File
Ticker(contract=CFD(conId=457068913, symbol='XAUUSD', exchange='SMART', currency='USD', localSymbol='XAUUSD', tradingClass='XAUUSD'))
Error 10168, reqId 4: Requested market data is not subscribed. Delayed market data is not enabled., contract: CFD(conId=457068913, symbol='XAUUSD', exchange='SMART', currency='USD', localSymbol='XAUUSD', tradingClass='XAUUSD')

**Reply by mattsta** [reply]
another thing to check: the market data is included "for free," but you still have to enable all the individual market data feeds (so they can track usage/agreements/requirements by upstream data vendors probably).

If you go to "Market Data Permissions" in the IBKR interface (web or mobile), do you have all the free data enabled? These are under the "Global" header:
![Screenshot 2022-09-17 at 10 36 16](https://user-images.githubusercontent.com/15594/190869523-66adb874-d454-42cf-bd94-e8ef35a2b329.png)

**Reply by brobbins107** [reply]
This worked! 

Settings -> Market Data Subscriptions -> Global -> Index CFD's on my UI. (and other free stuff included).
